debian/aeternity-node.service

```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Aeternity node
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Type=forking
+User=aeternity
+Group=aeternity
+ExecStart=/opt/aeternity/node/bin/aeternity start
+ExecStop=/opt/aeternity/node/bin/aeternity stop
+Restart=on-failure
```

Comment:
Could you please clarify your exploration and design of integration of `bin/aeternity start` with this init system?

Reply:
I guess your question is, is it working?  Tested with current code-base on a VM and no issues were observed. If you have suggestions for tests that we can run against the code running in systemd, they are welcome. 

I've used systemd with Erlang software before in production - the same approach `start`, `stop`. Generally there are no issues.  It is always a good idea to run a service from the init system than as a normal process.

Reply:
I see [ref](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=). May main question is whether this `on-failure` detects the failure. I.e. if you manually brutally kill the node process, does this service restart kick in?

Reply:
From what I remember by experience, I think it will. I think I also tested it with our code-base. I'll test it again and confirm.

Reply:
Investigations were made related to systemd and aeternity-node. What definitely is working is `service start`, `service stop`, `service restart`. Automatic restart on failure is not working. All possible variants for improvement were tested:
  * timeouts (systemd service file context)
 * running epmd as `ExecStartPre`
  * running epmd as the same user that runs aeternity-node; as a stand alone process
  * running epmd as a service (from erlang-base on Ubuntu 18.04)
  * running epmd as a service (custom service file)
  * disabling pre-start hooks (`--relx-disable-hooks` ; rule-out issues)

It all results in no automatic restart, or the node, not being able to start. All related to the epmd process. Sysytemd does not automatically restarts the service unless the epmd process is killed. If an aeternity sub-process is killed, only epmd is left in the state of the service and it is considered active. 

It should be working anyway compared with similar start scripts using Erlang. 

A separate issue will be open related to this matter.

Reply:
Thanks for checking.

> A separate issue will be open related to this matter.

Could you please confirm ticket tracking this.

Reply:
This one. https://www.pivotaltracker.com/n/projects/2152594/stories/165141235